charts/kubefed/charts/controllermanager/templates/roles.yaml

```diff
@@ -94,7 +96,9 @@ rules:
 - apiGroups:
   - core.kubefed.k8s.io
   resources:
-  - '*'
+  - federatedtypeconfigs
```

Comment:
Why is it necessary to desirable to be explicit to a the non-generated core group?  The approach you're proposing will require explicitly changing permissions when new types are added, which is something @xunpan and I tried to avoid in using `*`.

Reply:
This `Role` is specific to the admission webhook and aids in making the admission webhook run with least privileges. We already do not require access to all the core API types in the webhook, so I'm not sure we should enable access to all of them. Also, it doesn't seem too onerous to add an additional core type if/when we need to support defaulting or validating one in the webhook, especially since it's a non-generated group. Is it necessary to have the webhook wildcard this to all resources in the core group?

Reply:
In what sense does allowing the webhook access to all core types (validation for each of which should be mandatory going forward) something likely to violate the principle of least privilege?

Reply:
The webhook already doesn't validate all the resources in the group (e.g. `PropagatedVersion`, `ClusterPropagatedVersion`, `FederatedServiceStatus`), so by principle of least privilege, it should only be made to access the necessary resources. Using a wildcard would violate this today.